x-pack/plugins/security_solution/public/timelines/components/timeline/data_providers/provider_badge.tsx

```diff
@@ -10,53 +10,109 @@ import { isString } from 'lodash/fp';
 import React, { useCallback, useMemo } from 'react';
 import styled from 'styled-components';
 
+import { TimelineType } from '../../../../../common/types/timeline';
 import { getEmptyString } from '../../../../common/components/empty_value';
 import { ProviderContainer } from '../../../../common/components/drag_and_drop/provider_container';
 
-import { EXISTS_OPERATOR, QueryOperator } from './data_provider';
+import { DataProviderType, EXISTS_OPERATOR, QueryOperator } from './data_provider';
 
 import * as i18n from './translations';
 
-const ProviderBadgeStyled = (styled(EuiBadge)`
+type ProviderBadgeStyledType = typeof EuiBadge & {
+  timelineType: TimelineType;
```

Comment:
<img width="1680" alt="Screenshot 2020-07-01 at 10 47 10" src="https://user-images.githubusercontent.com/6295984/86230809-838a1600-bb89-11ea-8d0e-7b48cad27e1e.png">

I think we could change Line 22 into ```$timelineType: TimelineType```
according to https://styled-components.com/docs/api#transient-props
if we don't want passing `timelineType` down to EuiBadge

<img width="1663" alt="Screenshot 2020-07-01 at 10 55 11" src="https://user-images.githubusercontent.com/6295984/86230799-7ff68f00-bb89-11ea-9512-5a3f64107dbc.png">

Reply:
Nice!